app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Persistent\Test\Unit\CustomerData;
+
+use PHPUnit\Framework\TestCase;
+use Magento\Persistent\CustomerData\Persistent;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Helper\View;
+use Magento\Persistent\Helper\Session;
+use Magento\Persistent\Model\Session as PersistentSession;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+
+class PersistentTest extends TestCase
+{
+    /**
+     * @var Persistent
+     */
+    private $customerData;
+
+    /**
+     * @var Session|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $persistentSessionHelperMock;
+
+    /**
+     * @var View|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerViewHelperMock;
+
+    /**
+     * @var CustomerRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerRepositoryMock;
+
+    /**
+     * Setup environment for test
+     */
+    protected function setUp()
+    {
+        $this->persistentSessionHelperMock = $this->createMock(Session::class);
+        $this->customerViewHelperMock = $this->createMock(View::class);
+        $this->customerRepositoryMock = $this->createMock(CustomerRepositoryInterface::class);
+
+        $objectManager = new ObjectManagerHelper($this);
+
+        $this->customerData = $objectManager->getObject(
+            Persistent::class,
+            [
+                'persistentSession' => $this->persistentSessionHelperMock,
+                'customerViewHelper' => $this->customerViewHelperMock,
+                'customerRepository' => $this->customerRepositoryMock
+            ]
+        );
+    }
+
+    /**
+     * Test getSectionData() when disable persistent
+     */
+    public function testGetSectionDataWhenDisablePersistent()
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(false);
+
+        $this->assertEquals([], $this->customerData->getSectionData());
+    }
+
+    /**
+     * Test getSectionData() when customer doesn't login
+     */
+    public function testGetSectionDataWithNotLogin()
```

Comment:
In my personal opinion - the name of test should be more explicit - Optionally use pattern `given / when / then`. `testGetSectionDataWhenCustomerNotLoggedInReturnsEmptyArray`